Enable non-zero arity uninterpreted sorts in CVC4 backend

makaimann opened the pull request.

yoni206 reviewed (approved):
Great!
Maybe consider adding a test with arity -1, to make sure we get an exception.

makaimann commented:
Thanks for the review! Good idea, I added a test and fix with 7119e78aee3bdee085424c42bc92e22cbe570222. I was expecting CVC4 to complain if not provided enough parameters, but it doesn't so I just added the check myself.
